tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OboTests.cs

```diff
@@ -52,6 +53,180 @@ public void TestInitialize()
 
         #endregion
 
+        /// <summary>
+        /// Tests the behavior when calling OBO and silent in different orders with multiple users.
+        /// OBO calls should return tokens for correct users, silent calls should throw.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(false, false)]
+        [DataRow(true, false)]
+        [DataRow(true, true)]
+        public async Task OboAndSilent_Async(bool serializeCache, bool usePartitionedSerializationCache)
```

Comment:
nit / no action needed: integration tests are slow so we need to pick and choose scenarios where we use integration testing vs unit testing. Our unit testing strategy, where we fake out AAD, is pretty good (not perfect, needs more flexibility, but pretty good).

Reply:
Good point. When I was investigating these, I think I went with integration tests to make sure the E2E is correct and I don't mock something incorectly.